quic/client/state/ClientStateMachine.h

```diff
@@ -43,15 +45,21 @@ struct QuicClientConnectionState : public QuicConnectionStateBase {
   // TODO: use this to get rid of the data in the crypto stream.
   // folly::Optional<PacketNum> clientInitialPacketNum;
 
-  QuicClientConnectionState() : QuicConnectionStateBase(QuicNodeType::Client) {
+  QuicClientConnectionState(
+      std::shared_ptr<ClientHandshakeFactory> handshakeFactory_)
```

Comment:
Naming for `handshakeFactory_` is unconventional for us, something like `handshakeFactoryIn` would be more fitting.

Also the constructor should be marked `explicit`.